.github/workflows/release.yml

```diff
@@ -17,6 +17,47 @@ defaults:
     shell: bash
 
 jobs:
+  shellcheck:
+    runs-on: ubuntu-20.04
+    env:
+      SHELLCHECK_VERSION: '0.7.1'
+      SHELLCHECK_DIR: 'shellcheck'
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      # Retrieve annotated tags as the checkout above doesn't do this
+      # https://github.com/actions/checkout/issues/217#issuecomment-636481619
+      - name: Fetch Git Tag History
+        run: |
+          git fetch --depth=1 origin +refs/tags/*:refs/tags/*
+          echo "SHA=$(git rev-parse --short HEAD)" >> $GITHUB_ENV
+          echo "GIT_TAG_NAME=${GITHUB_REF#refs/tags/}" >> $GITHUB_ENV
+          echo "Current version: $(git describe)"
+
+      - name: Cache Shellcheck
+        id: restore-cache-sc
+        uses: actions/cache@v2
+        with:
+          path: shellcheck
+          # Stupid hack to invalidate cache from UI
+          # https://github.com/actions/cache/issues/2#issuecomment-673493515
+          key: sc-${{ env.SHELLCHECK_DIR }}-${{ env.SHELLCHECK_VERSION }}-${{ secrets.SC_CACHE_VERSION }}
+
+      - name: Download Shellcheck
+        if: steps.restore-cache-sc.outputs.cache-hit != 'true'
+        run: |
+          wget -qc https://github.com/koalaman/shellcheck/releases/download/v${{ env.SHELLCHECK_VERSION }}/shellcheck-v${{ env.SHELLCHECK_VERSION }}.linux.x86_64.tar.xz
```

Comment:
why not use `apt-get install shellcheck -y`?

Reply:
Probably, I guess I don't want to hit Ubuntu's servers if we can use our caching to restore the installation. But it's probably nothing to them and/or github has their own Ubuntu repo so they don't hit Ubuntu each time the developers do `apt-get`.

Reply:
No worries, they've mirrored all packages locally. Feel free to use apt-get here.